src/main/webapp/js/datahandlers/scholarshipdatahandler.js

```diff
@@ -19,33 +19,35 @@ const {Map: SoyMap} = goog.require('soy.map');
 const {addSpaceToCamelCase} = goog.require('datahandlers.utils');
 
 const NA = 'N/A';
-const REQUIREMENTS = ['academicRequirements', 
-                      'ethnicityRaceRequirements', 
-                      'financialRequirements', 
-                      'genderRequirements',
-                      'locationRequirements',
-                      'nationalOriginRequirements',
-                      'otherRequirements'];
+const REQUIREMENTS = [
+  'academicRequirements', 'ethnicityRaceRequirements', 'financialRequirements',
+  'genderRequirements', 'locationRequirements', 'nationalOriginRequirements',
+  'otherRequirements'
+];
 const SEPARATOR = ', ';
 const SCHOLARSHIP_ENDPOINT = '/scholarship-data';
 
-/** This class loads scholarship data from the backend and formats it for soy templates.  */
+/**
+ * This class loads scholarship data from the backend and formats it for soy
+ * templates.
+ */
 class ScholarshipDataHandler {
-
   /**
-   * This method converts from scholarship JSON object to a JS object map, 
+   * This method converts from scholarship JSON object to a JS object map,
    *  which will be used to render the scholarship page soy template.
    * @param {*} data - The JSON object to be converted.
    * @return {Object} - The object map representing a scholarship's data.
    * @private
    */
   async convertFromJsonToTemplate_(data) {
     const requirementsMap = new Map();
-                           
+
     let requirement = undefined;
     for (requirement of REQUIREMENTS) {
-      if (REQUIREMENTS[requirement] != undefined) {
-        requirementsMap.set(addSpaceToCamelCase(requirement), REQUIREMENTS[requirement].join(SEPARATOR));
+      if (data[requirement] != undefined) {
+        requirementsMap.set(
+            addSpaceToCamelCase(requirement),
+            data[requirement].join(SEPARATOR));
       } else {
         requirementsMap.set(addSpaceToCamelCase(requirement), NA);
```

Comment:
nit: don't substitute empty values. It clogs up your UI and pollutes the valuable text. I would leave them blank or just use a dash if you really want something there.